grakn-engine/src/main/java/ai/grakn/engine/backgroundtasks/taskstorage/GraknStateStorage.java

```diff
@@ -72,11 +73,11 @@ public String newState(String taskName, String createdBy, Date runAt, Boolean re
 
         Optional<String> result = attemptCommitToSystemGraph((graph) -> {
             InsertQuery query = graph.graql().insert(state);
-            String id = query.stream().findFirst().get().get(TASK_VAR).getId();
+            ConceptId id = query.stream().findFirst().get().get(TASK_VAR).getId();
 
             LOG.debug("Created " + graph.getConcept(id));
 
-            return id;
+            return id.getValue();
```

Comment:
The class `GraknStateStorage` extends `StateStorage` which has two implementations. the one based on using a grakn graph, and the other based on using an in memory map which uses only strings. This also ties in with a Zookeeper class which only uses Strings.

Basically putting everything on `ConceptId`feels wrong since only one implementation uses a Grakn Graph. There is definitely a way to improve it though. Generics on the interface is the way to go I think.